src/ESP8266MQTTMesh.cpp

```diff
@@ -388,13 +388,19 @@ void ESP8266MQTTMesh::scan() {
         ap_t *ap_last = NULL;
         for(ap_ptr = ap; ap_ptr != NULL; ap_last = ap_ptr, ap_ptr = ap_ptr->next) {
             if(network_idx >= 0) {
-                //AP is Wifi AP
-                if (ap_ptr->ssid_idx != NETWORK_MESH_NODE && rssi <= ap_ptr->rssi) {
+                //Tested Signal is Wifi AP
+                if ((ap_ptr->ssid_idx == NETWORK_MESH_NODE && //Current Signal is Mesh Node
+                     (rssi >= -80 || rssi >= ap_ptr->rssi))|| //and Signal under Test to direct AP have to be quite decent or at least better then current one
+                     ap_ptr->ssid_idx != NETWORK_MESH_NODE && rssi >= ap_ptr->rssi)//or both are Ap Points, but tested one have stronger Signal
+                {}else{
                    continue;
```

Comment:
I don't like this logic where you only perform action in the 'else'.  Please rework so that each condition actually performs an action directly and no 'else' is needed (or action is perfromed on both sides of the condition)